tests/wasmer-argus/src/argus/mod.rs

```diff
@@ -0,0 +1,302 @@
+        info!("fetching packages from {}", self.config.registry_url);
+
+        let m = MultiProgress::new();
+        let (s, mut r) = mpsc::unbounded_channel();
+
+        let mut pool = JoinSet::new();
+
+        {
+            let this = self.clone();
+            let bar = m.add(ProgressBar::new(0));
+
+            pool.spawn(async move { this.fetch_packages(s, bar).await });
+        }
+
+        let c = Arc::new(self.config.clone());
+
+        let mut count = 0;
+
+        let sem = Arc::new(Semaphore::new(self.config.jobs));
+
+        while let Some(pkg) = r.recv().await {
+            let c = c.clone();
+            let bar = m.add(ProgressBar::new(0));
+            let permit = Arc::clone(&sem).acquire_owned().await;
+
+            pool.spawn(async move {
+                let _permit = permit;
+                Argus::test(count, c, &pkg, bar).await
+            });
+
+            count += 1;
+        }
+
+        while let Some(t) = pool.join_next().await {
+            if let Err(e) = t {
+                error!("task failed: {e}")
+            }
+        }
+
+        info!("done!");
+        Ok(())
+    }
+
+    /// The actual test
+    async fn test(
+        test_id: u64,
+        config: Arc<ArgusConfig>,
+        pkg: &PackageVersionWithPackage,
+        p: ProgressBar,
+    ) -> anyhow::Result<()> {
+        p.set_style(
+            indicatif::ProgressStyle::with_template(&format!(
+                "[{test_id}] {{spinner:.blue}} {{msg}}"
+            ))
+            .unwrap()
+            .tick_strings(&["✶", "✸", "✹", "✺", "✹", "✷"]),
+        );
+
+        p.enable_steady_tick(Duration::from_millis(100));
+
+        let name = Argus::get_package_id(pkg);
+
+        let webc_url: Url = match &pkg.distribution.pirita_download_url {
+            Some(url) => url.parse().unwrap(),
+            None => {
+                info!("package {} has no download url, skipping", name);
+                p.finish_and_clear();
+                return Ok(());
+            }
+        };
+
+        p.set_message(format!("[{test_id}] testing package {name}",));
+
+        let path = Argus::get_path(config.clone(), pkg).await;
+        p.set_message(format!(
+            "testing package {name} -- path to download to is: {:?}",
+            path
+        ));
+
+        Argus::download_package(test_id, &path, &webc_url, &p).await?;
+
+        info!("package downloaded!");
+
+        p.reset();
+        p.set_style(
+            indicatif::ProgressStyle::with_template(&format!(
+                "[{test_id}] {{spinner:.blue}} {{msg}}"
+            ))
+            .unwrap()
+            .tick_strings(&["✶", "✸", "✹", "✺", "✹", "✷"]),
+        );
+
+        p.enable_steady_tick(Duration::from_millis(100));
+
+        p.set_message("package downloaded");
+        let filepath = path.join("package.webc");
+
+        let start = time::Instant::now();
+
+        let res = match std::panic::catch_unwind(|| {
```

Comment:
nit: a huge expression in a match is hard to read, I'd move this out to a variable first

Reply:
This and the other small changes: [6195830](https://github.com/wasmerio/wasmer/pull/4476/commits/619583065ca574c1ce0e958ffcb9c990af90eac6)